libdevcore/Common.h

```diff
@@ -37,6 +37,10 @@
 #pragma warning(disable:3682) //call through incomplete class
 #endif
 
+#ifdef __MIC__
```

Comment:
Please can you do a push/pop pair around only the problematic code and add a comment on exact which code the warning is being masked in?   Is this a boost header issue?